core/tx_pool_test.go

```diff
@@ -2561,3 +2571,342 @@ func BenchmarkPoolMultiAccountBatchInsert(b *testing.B) {
+	isLocal bool
+}
+
+const localIdx = 0
+
+func getTransactionGen(t *rapid.T, keys []*acc, nonces []uint64, localKey *acc, gasPriceMin, gasPriceMax, gasLimitMin, gasLimitMax uint64) *testTx {
+	idx := rapid.IntRange(0, len(keys)-1).Draw(t, "accIdx").(int)
+
+	var isLocal bool
+	var key *ecdsa.PrivateKey
+
+	if idx == localIdx {
+		isLocal = true
+		key = localKey.key
+	} else {
+		key = keys[idx].key
+	}
+
+	nonces[idx]++
+
+	gasPrice := big.NewInt(0).SetUint64(rapid.Uint64Range(gasPriceMin, gasPriceMax).Draw(t, "gasPrice").(uint64))
+
+	return &testTx{
+		tx:      pricedTransaction(nonces[idx]-1, rapid.Uint64Range(gasLimitMin, gasLimitMax).Draw(t, "gasLimit").(uint64), gasPrice, key),
+		isLocal: isLocal,
+	}
+}
+
+type transactionBatches struct {
+	batches   [][]*testTx
+	totalTxs  int
+	batchSize int
+}
+
+func transactionsGen(keys []*acc, nonces []uint64, localKey *acc, minTxs int, maxTxs int, gasPriceMin, gasPriceMax, gasLimitMin, gasLimitMax, minBatchSize uint64) func(t *rapid.T) transactionBatches {
+	return func(t *rapid.T) transactionBatches {
+		totalTxs := rapid.IntRange(minTxs, maxTxs).Draw(t, "totalTxs").(int)
+
+		minBatchSize := 10
+		if totalTxs < 10 {
+			minBatchSize = 1
+		}
+
+		batchSize := rapid.IntRange(minBatchSize, totalTxs).Draw(t, "batchSize").(int)
+
+		batchesN := totalTxs / batchSize
+		if totalTxs%batchSize > 0 {
+			batchesN++
+		}
+
+		batches := make([][]*testTx, batchesN)
+
+		keys = keys[:len(nonces)]
+
+		for i := 0; i < batchesN; i++ {
+			batches[i] = make([]*testTx, batchSize)
+
+			for j := 0; j < batchSize; j++ {
+				batches[i][j] = getTransactionGen(t, keys, nonces, localKey, gasPriceMin, gasPriceMax, gasLimitMin, gasLimitMax)
+			}
+		}
+
+		return transactionBatches{batches, totalTxs, batchSize}
+	}
+}
+
+func TestPoolBatchInsert(t *testing.T) {
+	t.Parallel()
+
+	const (
+		// the most tweakable params
+		gasLimit = 30_000_000
+
+		minTxs = 1_000
+		maxTxs = 10_000
+
+		minAccs = 10
+		maxAccs = maxTxs
+
+		minBatchSize = 10
+
+		// less tweakable, more like constants
+		gasPriceMin = 1
+		gasPriceMax = 1_000
+
+		gasLimitMin = params.TxGas
+		gasLimitMax = 30_000
+
+		balance = 1_000_000_000
+	)
+
+	initialBalance := big.NewInt(balance)
+
+	keys := make([]*acc, maxAccs)
+
+	var key *ecdsa.PrivateKey
+
+	// prealloc keys
+	for idx := 0; idx < maxAccs; idx++ {
+		key, _ = crypto.GenerateKey()
+
+		keys[idx] = &acc{
+			key:     key,
+			nonce:   0,
+			account: crypto.PubkeyToAddress(key.PublicKey),
+		}
+	}
+
+	rapid.Check(t, func(rt *rapid.T) {
+		fmt.Println("data generation started")
+
+		// Generate a batch of transactions to enqueue into the pool
+		testTxPoolConfig := testTxPoolConfig
+
+		// from sentry config
+		testTxPoolConfig.AccountQueue = 16
+		testTxPoolConfig.AccountSlots = 16
+		testTxPoolConfig.GlobalQueue = 32768
+		testTxPoolConfig.GlobalSlots = 32768
+		testTxPoolConfig.Lifetime = time.Hour + 30*time.Minute //"1h30m0s"
+		testTxPoolConfig.PriceLimit = 30000000000
```

Comment:
Can we get these configs from params directly ?

Reply:
Do we have a predefined config with that exact params?

Reply:
We can try getting from [here](https://github.com/maticnetwork/bor/blob/9dcc9935663a87425162d2d367c83f6d0bded2a8/internal/cli/server/config.go#L387).

Reply:
Why we introduced so many copy-paste types in cli/server ? like TxPoolConfig?

Reply:
Mostly to get a generic config, I guess.